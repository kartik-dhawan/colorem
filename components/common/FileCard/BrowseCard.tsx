import { Typography } from "@mui/material"
import EastIcon from "@mui/icons-material/East"
import GrainIcon from "@mui/icons-material/Grain"
import Image from "next/image"
import Link from "next/link"
import { useSelector } from "react-redux"
import { RootType } from "../../../redux/constants/stateTypes"

interface BrowseCardProps {
  fid: string
}

const BrowseCard = ({ fid }: BrowseCardProps) => {
  const content = useSelector((state: RootType) => state.contentSlice.data)

  return (
    <div className={fid + "RightFileContent"}>
      <div className={fid + "Content"}>
        <GrainIcon className={fid + "GrainIcon"} />
        <Typography variant="h5" className={fid + "ContentText"}>
          {content.browseCardContent}
        </Typography>
      </div>
      <div className={fid + "RightFileImage"}>
        <Image
          alt="bitmoji image"
          height={425}
          width={425}
          src="/images/browse_bitmoji.webp"
          priority
        />
      </div>
      <Link href="/palettes" className={fid + "ArrowIconWrapper"}>
        <div className={fid + "Browse"}>Browse more</div>
        <EastIcon className={fid + "ArrowIcon"} />
      </Link>
    </div>
  )
}

export default BrowseCard
