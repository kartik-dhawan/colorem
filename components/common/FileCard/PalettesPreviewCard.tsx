import { API_URLS } from "../../../utils/constants"
import { PaletteDataType } from "../../../utils/interfaces"
import { fetcher, getShuffledArray } from "../../../utils/methods"
import useSWR from "swr"
import { Box, Container, Typography } from "@mui/material"
import Link from "next/link"
import { useId } from "react"

interface PalettesPreviewProps {
  fid: string
}

const PalettesPreviewCard = ({ fid }: PalettesPreviewProps) => {
  const id = useId()

  const { data, error, isLoading } = useSWR<PaletteDataType[]>(
    API_URLS.GET_ALL_PALETTES,
    fetcher
  )

  // const randomPalettes = data && getShuffledArray(data, 4)

  return (
    <>
      {isLoading ? (
        <div>loading....</div>
      ) : !error && data ? (
        <Container
          className={fid + "PreviewCardWrapper"}
          id={fid + "PreviewCardWrapper"}
          sx={{
            display: "flex",
            height: "100%",
            flexDirection: "column",
            justifyContent: "space-around",
            padding: "2rem 0px",
            position: "relative",
          }}
        >
          {getShuffledArray(data, 4)?.map((palette: PaletteDataType) => {
            console.log(palette)
            return (
              <Box
                key={palette._id}
                className={fid + "PreviewPaletteWrapper"}
                id={id + fid + "PreviewPaletteWrapper"}
              >
                <Box
                  className={fid + "PreviewPaletteColorsWrapper"}
                  id={id + fid + "PreviewPaletteColorsWrapper"}
                  sx={{
                    display: "flex",
                    height: "56px",
                  }}
                >
                  {palette.hex.map((hex: string) => {
                    return (
                      <Box
                        key={id + hex}
                        className={fid + "PreviewPaletteColor"}
                        id={id + fid + "PreviewPaletteColor"}
                        sx={{
                          backgroundColor: `#${hex}`,
                          height: "100%",
                          flexGrow: 1,
                          transition: "200ms all ease",
                          "&:hover": {
                            flexGrow: 3,
                          },
                        }}
                      ></Box>
                    )
                  })}
                </Box>
                <Typography
                  className={fid + "PreviewPaletteName"}
                  id={id + fid + "PreviewPaletteName"}
                  variant="body1"
                >
                  {palette.name}
                </Typography>
              </Box>
            )
          })}
          <Link href="/palettes">
            <Typography
              sx={{
                position: "absolute",
                right: 0,
                bottom: 0,
                color: "#d9d9d9",
                borderBottom: "1px solid #c4c4c4",
              }}
            >
              View more
            </Typography>
          </Link>
        </Container>
      ) : (
        <div>Error</div>
      )}
    </>
  )
}

export default PalettesPreviewCard
