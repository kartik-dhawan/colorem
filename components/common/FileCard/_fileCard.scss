// file-cards css
.fileSectionCard {
  .fileSectionCardLower {
    position: relative;
    height: 500px;
    @include mobiles-sm {
      height: 300px;
    }
    @include mobiles {
      height: 350px;
    }
    @include tablets {
      height: 400px;
    }
    .fileSectionRightFileContent {
      height: 100%;
      display: flex;
      justify-content: space-between;
      align-items: flex-end;
      .fileSectionContent {
        height: 100%;
        z-index: 10;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        @include desktops {
          width: 25%;
        }
        @include tablets-landscape {
          width: 30%;
        }
        @include tablets-portrait {
          flex-direction: row;
        }
        @include mobiles {
          flex-direction: row;
        }
        .fileSectionGrainIcon {
          color: $text_secondary;
          margin: 20px;
          font-size: 40px;
          cursor: pointer;
          @include mobileAndTablet {
            font-size: 32px;
          }
        }
        .fileSectionContentText {
          padding: 0px 0px 20px 25px;
          color: $text_secondary;
          font-size: 30px;
          @include tablets {
            font-size: 22px;
          }
          @include between-tablets-and-desktop {
            font-size: 27px;
          }
          @include mobiles {
            font-size: 21px;
          }
          @include tablets-portrait {
            padding: 20px 0px;
            width: max-content;
            position: absolute;
            left: 12%;
          }
          @include mobiles {
            padding: 20px 15px;
            position: absolute;
            left: 15%;
            font-size: 20px;
          }
          @include mobiles-sm {
            padding: 16px 15px;
            position: absolute;
            left: 17%;
            font-size: 19px;
          }
        }
      }
      .fileSectionRightFileImage {
        position: absolute;
        width: 100%;
        img {
          position: relative;
          left: calc(50% - 1.5rem);
          transform: translate(-50%);
          margin-bottom: -4px;
          @include mobiles-sm {
            width: 235px;
            height: 235px;
          }
          @include mobiles {
            width: 275px;
            height: 275px;
          }
          @include tablets {
            width: 300px;
            height: 300px;
          }
          @include between-tablets-and-desktop {
            width: 375px;
            height: 375px;
          }
        }
      }
      .fileSectionArrowIconWrapper {
        border: 2px solid $text_secondary;
        display: flex;
        justify-content: space-between;
        margin: 20px 20px;
        padding: 10px;
        border-radius: 1000px;
        cursor: pointer;
        z-index: 10;
        color: $text_primary;
        text-decoration: none;
        @include tablets {
          padding: 7px;
        }
        @include tablets-up {
          width: 25%;
        }
        .fileSectionBrowse {
          text-align: left;
          align-self: center;
          margin: 0 6%;
          font-size: 20px;
          color: $text_secondary;
          overflow: hidden;
          white-space: nowrap;
          text-overflow: ellipsis;
          font-weight: 400;
          &:any-link {
            text-decoration: none;
          }
          @include mobiles {
            display: none;
          }
          @include mobiles-sm {
            display: none;
          }
        }
        .fileSectionArrowIcon {
          background-color: $background_primary;
          padding: 12px;
          border-radius: 1000px;
          @include tablets {
            padding: 6px;
          }
        }
      }
    }
  }
}
