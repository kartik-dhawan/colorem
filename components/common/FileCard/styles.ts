export const styles = {
  // folder shape containers
  fileSectionCardLowerRight: {
    borderRadius: {
      xs: "30px 0px 30px 30px",
      md: "30px 0px 30px 30px",
    },
    marginRight: "1.5rem",
    background: "linear-gradient(180deg, #19c7a20d 0%, #51bdd938 90%)", // green - green_color_1, green_color_2
    boxShadow: "0 2px 56px -18px #8787875c",
    backdropFilter: "blur( 4px )",
    height: {
      xs: "335px",
      sm: "350px",
      md: "400px",
      lg: "500px",
    },
  },
  fileSectionCardLowerLeft: {
    borderRadius: {
      xs: "0px 30px 30px 30px",
      md: "0px 30px 30px 30px",
    },
    marginLeft: "1.5rem",
    minHeight: {
      xs: "335px",
      sm: "350px",
      md: "400px",
      lg: "500px",
    },
    height: "max-content",
    border: "1px solid #fff3",
    boxShadow: "0 2px 56px -18px #8787875c",
    backdropFilter: "blur( 4px )",
  },
}
