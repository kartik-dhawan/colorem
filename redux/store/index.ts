import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
