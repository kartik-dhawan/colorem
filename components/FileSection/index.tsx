import { Grid } from "@mui/material"
import FileCard from "../common/FileCard"

const FileSection = () => {
  const fid = "fileSection"

  return (
    <Grid
      container
      className={fid + "Wrapper"}
      direction="row"
      data-testid={fid + "Wrapper"}
    >
      <FileCard type="right" fid={fid} fr={8} />
      <FileCard type="left" fid={fid} fr={4} />
    </Grid>
  )
}

export default FileSection
