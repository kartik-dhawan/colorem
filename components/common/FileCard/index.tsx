import { Grid } from "@mui/material"
import { styles } from "./styles"
import BrowseCard from "./BrowseCard"
import React from "react"
import PalettesPreviewCard from "./PalettesPreviewCard"

interface FileCardType {
  type: string
  fid: string
  fr: number
}

const FileCard = ({ type, fid, fr }: FileCardType) => {
  return (
    <Grid
      item
      md={fr}
      xs={12}
      className={fid + "Card"}
      sx={{
        marginBottom: "30px",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <Grid
        item
        className={fid + "CardLower"}
        id={fid + "CardLower"}
        sx={
          type === "right"
            ? styles.fileSectionCardLowerRight
            : styles.fileSectionCardLowerLeft
        }
      >
        {/* conditionally rendered card content */}
        {type === "right" && <BrowseCard fid={fid} />}
        {type === "left" && <PalettesPreviewCard fid={fid} />}
      </Grid>
    </Grid>
  )
}

export default FileCard
