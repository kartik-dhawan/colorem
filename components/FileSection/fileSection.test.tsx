import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import FileSection from "."
import store from "../../redux/store"
import { render } from "@testing-library/react"
import FileCard from "../common/FileCard"

describe("File section Component", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <FileSection />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- should render right file card correctly", () => {
    const view = render(
      <Provider store={store}>
        <FileCard type="right" fid="fileSection" fr={8} />
      </Provider>
    )
    const fileSectionRightCard = view.getByTestId("fileSectionright")
    expect(fileSectionRightCard).toBeInTheDocument()
  })

  it("-- should render left file card correctly", () => {
    const view = render(
      <Provider store={store}>
        <FileCard type="left" fid="fileSection" fr={4} />
      </Provider>
    )
    const fileSectionLeftCard = view.getByTestId("fileSectionleft")
    expect(fileSectionLeftCard).toBeInTheDocument()
  })
})
