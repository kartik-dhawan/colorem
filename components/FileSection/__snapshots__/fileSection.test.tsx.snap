// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File section Component -- should match snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container fileSectionWrapper css-11lq3yg-MuiGrid-root"
  data-testid="fileSectionWrapper"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 fileSectionCard css-k0o1dg-MuiGrid-root"
    data-testid="fileSectionright"
  >
    <div
      className="MuiGrid-root MuiGrid-item fileSectionCardLower css-1hjlv0p-MuiGrid-root"
      id="fileSectionCardLower"
    >
      <div
        className="fileSectionRightFileContent"
      >
        <div
          className="fileSectionContent"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium fileSectionGrainIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="GrainIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 12c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zM6 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12-8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm-4 8c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-4-4c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
            />
          </svg>
          <h5
            className="MuiTypography-root MuiTypography-h5 fileSectionContentText css-ag7rrr-MuiTypography-root"
          />
        </div>
        <div
          className="fileSectionRightFileImage"
        >
          <img
            alt="bitmoji image"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height={425}
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=1080&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fbrowse_bitmoji.webp&w=1080&q=75 2x"
            style={
              {
                "color": "transparent",
              }
            }
            width={425}
          />
        </div>
        <a
          className="fileSectionArrowIconWrapper"
          href="/gradients"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="fileSectionBrowse"
          >
            Browse
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium fileSectionArrowIcon css-i4bv87-MuiSvgIcon-root"
            data-testid="EastIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 fileSectionCard css-1eqpqzh-MuiGrid-root"
    data-testid="fileSectionleft"
  >
    <div
      className="MuiGrid-root MuiGrid-item fileSectionCardLower css-5w37io-MuiGrid-root"
      id="fileSectionCardLower"
    >
      <div
        className="MuiContainer-root MuiContainer-maxWidthLg css-r2ks57-MuiContainer-root"
      >
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-4i0lvv-MuiSkeleton-root"
          style={
            {
              "height": 56,
              "width": "100%",
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-ovvgps-MuiSkeleton-root"
          style={
            {
              "height": 56,
              "width": "100%",
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-aedbwi-MuiSkeleton-root"
          style={
            {
              "height": 56,
              "width": "100%",
            }
          }
        />
        <span
          className="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-2ql2a-MuiSkeleton-root"
          style={
            {
              "height": 56,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
